src/main/java/org/broadinstitute/hellbender/tools/walkers/mutect/filtering/PolymorphicNuMTFilter.java

```diff
@@ -29,16 +30,34 @@ public PolymorphicNuMTFilter(final double medianAutosomalCoverage){
     public ErrorType errorType() { return ErrorType.NON_SOMATIC; }
 
     @Override
-    public boolean isArtifact(final VariantContext vc, final Mutect2FilteringEngine filteringEngine) {
-        return vc.getGenotypes().stream().filter(filteringEngine::isTumor)
-                .filter(Genotype::hasAD)
-                .anyMatch(g -> {
-                    final int[] alleleDepths = g.getAD();
-                    final int numRealAlleles = vc.hasSymbolicAlleles() ? alleleDepths.length - 1 : alleleDepths.length;
-                    //Start at first alternate allele depth (the ref allele is first)
-                    final OptionalInt max = IntStream.range(1, numRealAlleles).map(a -> alleleDepths[a]).max();
-                    return max.getAsInt() < maxAltDepthCutoff;
-                });
+    public Predicate<Genotype> getPredicateForData() {
+        return Genotype::hasAD;
+    }
+
+    public List<Integer> getDataForAlleles(Genotype g) {
+        return Arrays.stream(g.getAD()).boxed().collect(Collectors.toList());
+    }
+
+    @Override
+    public Double getDefaultProbability() {
+        return Double.NaN;
+    }
+
+
+    @Override
+    public Map<Allele, Boolean> areAllelesArtifacts(final LinkedHashMap<Allele, List<Integer>> dataByAllele, final VariantContext vc, final Mutect2FilteringEngine filteringEngine, ReferenceContext referenceContext) {
+        Map<Allele, Boolean> results = new HashMap<>();
+        final int numAlleles = dataByAllele.size();
+        final int numRealAlleles = vc.hasSymbolicAlleles() ? numAlleles - 1 : numAlleles;
```

Comment:
Are symbolic alleles always last? If not, I'm not sure the loop below works correctly.

Reply:
We do that in GATK, but it's not in the spec, so I don't think it's a safe assumption.